### PR TITLE
Oxidation replace message

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    ) -> PyResult<PyObject> {
+        let msg = libparsec::types::MessageContent::decrypt_verify_and_load_for(
+            ciphered,
+            &recipient_privkey.0,
+            &author_verify_key.0,
+            &expected_author.0,
+            expected_timestamp.0,
+        );
+
+        Ok(unsafe {
```
Comment (FirelightFlagboy) [inline src/message.rs:76]: No way to remove the use of `unsafe` ?

Comment (TimeEngineer) [reply src/message.rs:76]: Nope if we work with subclass

Comment (FirelightFlagboy) [reply src/message.rs:76]: Add a comment to explain why this block is safe, I plan to enable [`clippy::undocumented_unsafe_blocks`](https://rust-lang.github.io/rust-clippy/master/#undocumented_unsafe_blocks).
Can you also limit the size of the unsafe block to code that required ?

i.e.: only put `into_new_object` & `from_owned_ptr` into the `unsafe` block and keep everything else outside

File: src/message.rs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+                let initializer = PyClassInitializer::from((PingMessageContent, Self(msg)));
+                // This block is safe because `initializer` has the correct type
+                unsafe {
+                    let ptr =
+                        initializer.into_new_object(py, PingMessageContent::type_object_raw(py))?;
+                    PyObject::from_owned_ptr(py, ptr)
+                }
+            }
+            Err(err) => return Err(PyValueError::new_err(err)),
+        })
```
Comment (FirelightFlagboy) [inline src/message.rs:119]: Is it worth to use a macro here, to factorize the code ?

File: src/message.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply src/message.rs]: :+1:

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**